modin/logging/logger_decorator.py

```diff
@@ -0,0 +1,143 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""
+Module contains ``logger_decorator`` function.
+
+``logger_decorator`` is used for decorating individual Modin functions.
+"""
+
+from typing import Optional, Callable, Union, Type
+from types import FunctionType, MethodType
+from functools import wraps
+from logging import Logger
+
+from modin.config import LogMode
+from .config import get_logger
+
+_MODIN_LOGGER_NOWRAP = "__modin_logging_nowrap__"
+
+
+def disable_logging(func):
+    """
+    Disable logging of one particular function. Useful for decorated classes.
+
+    Parameters
+    ----------
+    func : callable
+        A method in a logger-decorated class for which logging should be disabled.
+
+    Returns
+    -------
+    func
+        A function with logging disabled.
+    """
+    setattr(func, _MODIN_LOGGER_NOWRAP, True)
+    return func
+
+
+def logger_decorator(
```

Comment:
Sorry for my pickiness on the names. Should this be just `log` or `enable_logging`?

Reply:
`log` is too general, but maybe renaming to `enable_logging` could make sense (as I already have a `disable_logging` counterpart)...